### PR TITLE
[doctor command] Add tests for androidSDK and androidNDK

### PR DESCRIPTION
**Pull request by thib92**
Summary:
---------

Add unit tests for the healthchecks for `androidSDK` and `androidNDK`

Also removed a `$FlowFixMe` that wasn't removed when the `androidSDK` file was moved to TS.

Related to #694 

Test Plan:
----------

CI is green

### CODE REVIEW

**Comment by thib92** [comment]
@lucasbento tests are failing on Windows because apparently the `androidSDK` check specific to Windows is not responding within 5s.

I'm not too sure why but I'm suspecting the `execa` call here is not responding. Should we put a timeout on it?

https://github.com/react-native-community/cli/blob/ed0a6fd419dba778e5ea737a040092afdda51c47/packages/cli/src/commands/doctor/healthchecks/androidSDK.ts#L24-L29

**Comment by lucasbento** [comment]
@thib92: I would say so, it seems like a command that can take quite a while to respond.

I'm not sure if 5 seconds would be sufficient though, @thymikee can you tell us about that?

I didn't actually do this piece of code so I am not sure about it.

**Comment by thymikee** [comment]
I'd mock a response that we expect, no time for long timeouts.

**Comment by lucasbento** [comment]
Yes, that's definitely a better approach 😄

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidNDK.test.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import androidNDK from '../androidNDK';
+import getEnvironmentInfo from '../../../../tools/envinfo';
+import {EnvironmentInfo} from '../../types';
+import {NoopLoader} from '../../../../tools/loader';
+
+import * as common from '../common';
+
+const logSpy = jest.spyOn(common, 'logManualInstallation');
+
+describe('androidHomeEnvVariables', () => {
```
Comment (lucasbento) [inline packages/cli/src/commands/doctor/healthchecks/__tests__/androidNDK.test.ts]: Need to change this to `androidNDK`.

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidNDK.test.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('returns a message if the Android SDK is not installed', async () => {
+    environmentInfo.SDKs['Android SDK'] = 'Not Found';
+    const diagnostics = await androidNDK.getDiagnostics(environmentInfo);
+    expect(typeof diagnostics.needsToBeFixed).toBe('string');
```
Comment (lucasbento) [inline packages/cli/src/commands/doctor/healthchecks/__tests__/androidNDK.test.ts]: This has actually been changed on #743, you can check the change for `androidHomeEnvVariable` [here](https://github.com/react-native-community/cli/pull/743/files#diff-9c043f20be368d1006196f0b0cb243c2R24).

Comment (lucasbento) [reply packages/cli/src/commands/doctor/healthchecks/__tests__/androidNDK.test.ts]: Just to clarify, this field was being used to hold a boolean and a string, the string was used as a `description` sorta of thing, on #743 I actually created the `description` field.

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import androidSDK from '../androidSDK';
+import getEnvironmentInfo from '../../../../tools/envinfo';
+import {EnvironmentInfo} from '../../types';
+import {NoopLoader} from '../../../../tools/loader';
+
+import * as common from '../common';
+
+const logSpy = jest.spyOn(common, 'logManualInstallation');
+
+describe('androidHomeEnvVariables', () => {
```
Comment (lucasbento) [inline packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: Need to change this to `androidSDK`.

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    environmentInfo.SDKs['Android SDK'] = 'Not Found';
+    const diagnostics = await androidSDK.getDiagnostics(environmentInfo);
+
+    jest.mock('execa', () => ({
+      default: () => {
+        throw new Error('No SDK found');
+      },
+    }));
+
+    expect(typeof diagnostics.needsToBeFixed).toBe('string');
```
Comment (lucasbento) [inline packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: Same as the other one.

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('returns a message if the Android SDK is not installed', async () => {
+    environmentInfo.SDKs['Android SDK'] = 'Not Found';
+    const diagnostics = await androidSDK.getDiagnostics(environmentInfo);
+
+    jest.mock('execa', () => ({
```
Comment (lucasbento) [inline packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: I suggest mocking this at the top of the file, you can do:

```js
jest.mock('execa', () => ({
  default: jest.fn(),
});
```

And then, on each test case:

```js
execa.mockReturnValueOnce('my value');
```

I believe this would work.

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        return 'build-tools;25';
+      },
+    }));
+
+    const diagnostics = await androidSDK.getDiagnostics(environmentInfo);
+    expect(typeof diagnostics.needsToBeFixed).toBe('string');
+  });
+
+  it('returns false if the SDK version is in range', async () => {
+    // @ts-ignore
```
Comment (lucasbento) [inline packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: Why is this needed?

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+      default: () => {
+        return 'build-tools;27';
+      },
+    }));
+
+    const diagnostics = await androidSDK.getDiagnostics(environmentInfo);
+    expect(diagnostics.needsToBeFixed).toBe(false);
+  });
+
+  it('logs manual installation steps to the screen', async () => {
```
Comment (lucasbento) [inline packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: This doesn't have to be `async` as it's not `await`ing anything.

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts (hunk 5 above)
Comment (thib92) [reply packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: Thanks! That worked 😄 
I'm not an export on Jest mocks 🙄

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts (hunk 7 above)
Comment (thib92) [reply packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: Good catch!

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts (hunk 6 above)
Comment (thib92) [reply packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: Because the `Android SDK` key of envinfo needs more than just `Build Tools`, but I didn't want to fake version for everything it provides, as they are not needed by the `androidSDK` healthcheck.

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidNDK.test.ts (hunk 2 above)
Comment (thib92) [reply packages/cli/src/commands/doctor/healthchecks/__tests__/androidNDK.test.ts]: Okay nice one. I like it like that!

**Comment by thib92** [comment]
@lucasbento thanks for the good review. Everything should be fixed.

**Comment by thib92** [comment]
Btw I'm doing a lot of copy-paste between all the healthcheck tests. Once I'm done with all of them, I'll see if I can create util functions to avoid repeating code in the tests.

**Comment by lucasbento** [comment]
@thymikee: can you give a review?

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+import getEnvironmentInfo from '../../../../tools/envinfo';
+import {EnvironmentInfo} from '../../types';
+import {NoopLoader} from '../../../../tools/loader';
+
+import * as common from '../common';
+
+const logSpy = jest.spyOn(common, 'logManualInstallation');
+
+const mockExeca = (stdout: string) => {
+  jest.mock('execa', () => jest.fn());
```
Comment (thymikee) [inline packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: `jest.mock` is hosited to the top of the parent scope, it's not meant to be called mutliple times. Can you move there so it's a bit more obvious the `execa` _module_ is mocked?

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+import androidSDK from '../androidSDK';
+import getEnvironmentInfo from '../../../../tools/envinfo';
+import {EnvironmentInfo} from '../../types';
+import {NoopLoader} from '../../../../tools/loader';
+
+import * as common from '../common';
+
+const logSpy = jest.spyOn(common, 'logManualInstallation');
+
+const mockExeca = (stdout: string) => {
```
Comment (thymikee) [inline packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: Let's rename it to `mockSdkManagerResponse` or something so that it's closer to what it's actually doing, since we don't extract execa call to a function we could mock – which would make it obvious what is mocked.

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts (hunk 8 above)
Comment (thib92) [reply packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: Changed it a little bit to be easier to understand and more standard. What do you think?

File: packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts (hunk 9 above)
Comment (thib92) [reply packages/cli/src/commands/doctor/healthchecks/__tests__/androidSDK.test.ts]: See what I changed. What do you think?
